docs/txp.md

```diff
@@ -0,0 +1,486 @@
+
+Witness `ScriptWitness` contains:
+
+* validator script, `validator`
+* redeemer script, `redeemer`
+
+We require:
+
+* Address is a `Script` address
+* Address spending data is created with validator script same as `validator`
+* Both `redeemer` and `validator` scripts have the same version.
+* If `verifyAllIsKnown` is `True`, script version (which is same for
+    redeemer and validator) must be known.
+    - For Cardano SL 1.0 it means that version must be equal to 0,
+      it's the only version known to our software.
+* Plutus validation script built from redeemer and validator for
+    given tx must return `True`
+
+###### Redeem witness
+
+Witness `PkWitness` contains:
+
+* redeem public key `pk`
+* redeem signature `sig`
+
+We require:
+
+
+* Address is a `Redeem` address
+* Address spending data is created with redeem public key `pk`
+* Signature `sig` is a valid signature of transaction (validity
+  checked with `pk`) As in public key witness, signature inside redeem
+  witness must be given for `TxSigData` (which is basically the same
+  as `Hash Tx`). `SignRedeemTx` tag should be. The sequence of bytes
+  to sign is: `02 <magic> HASH_CBOR` (it differs
+  from [Public key witness](#public-key-witness) in only one byte).
+
+
+###### Unknown witness
+
+* If `verifyAllIsKnown` is `True`, witness is invalid
+* If witness has unknown type with tag `t`, corresponding address must
+  have unknown type with tag `t`
+
+##### Sums check
+
+This check is omitted if (and only if):
+* There is an unknown transaction input
+
+For each input and output we know its value (i.e. associated balance). We compute sum of all inputs and all outputs.
+
+* If sum of outputs is greater than sum of inputs, tx is considered invalid
+* Otherwise, check succeeds, difference between sum of inputs and sum
+  of outputs is considered to be *transaction fee*.
+
+##### Transaction fee check
+
+This check is omitted if (and only if) one of conditions met:
+* There is an unknown transaction input
+* All inputs correspond to `Redeem` addresses
+  - Implication of this rule is that one can redeem ADA without paying fee
+* Currently adopted tx fee policy is unknown
+
+For this check we know:
+
+* Transaction fee `txFee` (from *Sums check*)
+* Transaction size `txSize` (from *Transaction size check*)
+* Currently adopted tx fee policy (as part of `BlockVersionData`), which is of known type
+
+###### Size-linear policy
+Only policy supported by Cardano SL v1.0 is `TxFeePolicyTxSizeLinear`.
+
+For it, we need to compute minimal required fee for this transaction.
+This value is computed using fixed-precision arithmetic (precision is 1e-9) using formula:
+
+```
+minFee = a + b * txSize
+```
+
+There are two cases check fails (and transaction rejected):
+
+* Minimal fee is negative or is greater than maximal number of coins in the system
+
+   ```minFee < 0 || minFee > maxBound @Coin```
+   - It probably indicates there is a mistake in currently adopted fee
```

Comment:
Why do you use bullet-list consisting of the one item? Why not to just write it without bullet label?

Reply:
It's better than putting one sentence. Bullet lust means adding more details to broader point, which is the case here